File: seeker/snippet/Ternary.java
//date: 2022-10-06T17:09:04Z
//url: https://api.github.com/gists/ce8d399c763a3824f6d6e862993e8365
//owner: https://api.github.com/users/LisTata

package com.company;

import java.util.Scanner;

public class Ternary {
    public static void main(String[] args) {
        //Продажа табачных изделий до 18 лет запрещена
        int age= 18;
        System.out.println((age>=18 ? "Yes": "No"));
        //Строчная или заглавная
        System.out.printf("%d, %d, %d, %d, %n ", (int)'a', (int)'z',(int)'A', (int)'Z');
        char ch='b';
        String answer = (int)ch>=97 && (int)ch<=122 ? "lowercase":(int)ch>=65 &&(int)ch<=90?"uppercase":"not a letter";
        System.out.println(answer);

        //Наибольшее из 3 чисел.
        int a=1, b=3, c=2;
        //int max = a>b? b>c? a:a>c? a:c:a>c? b: b>c? b:c;
        int max=a>b && a>c? a:
                b>a && b>c? a:
                c>a && c>b? c: 0;
        int a2 = (int)(Math.random()*100+1);
       // System.out.println(a2);
        int b2=0;
        Scanner sc = new Scanner(System.in);

        while (a2!=b2){
            System.out.println("Enter number: ");
             b2 =sc.nextInt();
             if(a2>b2)
                 System.out.println("больше");
             else
                 System.out.println("меньше");

        }
        System.out.println("Win");







    }
}

File: seeker/snippet/ternary.go
//date: 2025-03-26T16:55:34Z
//url: https://api.github.com/gists/a4786eb2dff87b4e62d1d711da77221d
//owner: https://api.github.com/users/ttys3

package ternary

func If[T any](condition bool, ifTrue Supplier[T], ifFalse Supplier[T]) T {
	if condition {
		return ifTrue.Get()
	}
	return ifFalse.Get()
}

func Func[T any](supplier func() T) Supplier[T] {
	return FuncSupplier[T](supplier)
}

func Value[T any](value T) Supplier[T] {
	return ValueSupplier[T]{Value: value}
}

type Supplier[T any] interface {
	Get() T
}

type FuncSupplier[T any] func() T

func (s FuncSupplier[T]) Get() T {
	return s()
}

type ValueSupplier[T any] struct {
	Value T
}

func (s ValueSupplier[T]) Get() T {
	return s.Value
}
